lib/hooks/api/useFetchMetricStats.ts

```diff
@@ -0,0 +1,41 @@
+import useSWR, { Fetcher } from "swr";
+import { publicApiFetcher } from "lib/utils/public-api-fetcher";
+
+type UseFetchMetricStatsProps = {
```

Comment:
refactor (blocking): the type is for params, not props of a component.

```suggestion
type UseFetchMetricStatsParams = {
```